<launch>
  <rosparam param="/use_sim_time">true</rosparam>

  <node name="esvio_image_representation_left" pkg="esvio_image_representation" type="esvio_image_representation" output="screen">
    <remap from="events" to="/davis/left/events" />
    <remap from="imu" to="/davis/left/imu" />
    <remap from="image" to="/davis/left/image_raw" />
    <remap from="camera_info" to="/davis/left/camera_info" />
    <remap from="image_representation" to="image_representation_left" />
    <remap from="image_representation_temp" to="image_representation_temp_left" />
    <rosparam command="load" file="$(find esvio_image_representation)/cfg/esvio_image_representation_l.yaml" />
  </node>

  <node name="esvio_image_representation_right" pkg="esvio_image_representation" type="esvio_image_representation" output="screen">
    <remap from="events" to="/davis/right/events" />
    <remap from="imu" to="/davis/right/imu" />
    <remap from="image" to="/davis/right/image_raw" />
    <remap from="camera_info" to="/davis/right/camera_info" />
    <remap from="image_representation" to="image_representation_right" />
    <remap from="image_representation_temp" to="image_representation_temp_right" />
    <rosparam command="load" file="$(find esvio_image_representation)/cfg/esvio_image_representation_r.yaml" />
  </node>

  <node name="global_timer" pkg="rostopic" type="rostopic" args="pub -s -r 50 /sync std_msgs/Time 'now' ">
    <rosparam param="/use_sim_time">true</rosparam>
  </node>

  <!-- rqt_gui -->
  <node pkg="rqt_gui" type="rqt_gui" name="rqt_gui" args="--perspective-file $(find esvio_image_representation)/esvio_image_representation.perspective" />

  <node pkg="rosbag" type="play" name="play" output="screen"
    args="/home/njk/data/dsec/zurich_city_04_a_imu.bag -r 0.5 -q  --clock">
  </node>

  <include file="$(find esvio_image_representation)/launch/camera_info_dsec.launch" />
</launch>